src/number/format-decimal.js

```diff
@@ -42,7 +51,11 @@ module.exports = function(value, localeData, options) {
 
 		digitNum++;
 
-		if (digitNum > 1 && digitNum % localeData.groupSize === 1) {
+		if (digitNum > 1 && digitNum % currentGroupSize === 1) {
+			if ( groupSizes.length > 0) {
+				currentGroupSize = groupSizes.shift();
+			}
+			digitNum = 1;	// reset this back to 1 to handle variable group sizes
```

Comment:
This caught me off guard as well. But because line 54 has `digiNum % currentGroupSize ===1`, this 'counter' should start at 1 as well. For example, for group sizes of 3, you want to the code to run on the 4th,7th, and 10th digits.
If we set digitNum=0, it would run on the 4th, 8th and 12th. 
Would it be more readable if I changed `digitNum = 1` to `digitNum = digitNum - currentGroupSize`?